### PR TITLE
ARROW-2070: [Python] Fix chdir logic in setup.py

### PR DESCRIPTION
**Pull request by pitrou**
In some conditions, setup.py may change the current directory and omit restoring the previous one, which can fail some operations.

### CODE REVIEW

File: python/setup.py [hunk 1]
```diff
@@ -129,162 +141,160 @@ def _run_cmake(self):
         # The staging directory for the module being built
         build_temp = pjoin(os.getcwd(), self.build_temp)
         build_lib = os.path.join(os.getcwd(), self.build_lib)
-
-        # Change to the build directory
         saved_cwd = os.getcwd()
+
         if not os.path.isdir(self.build_temp):
             self.mkpath(self.build_temp)
-        os.chdir(self.build_temp)
```
Comment (pitrou) [inline python/setup.py]: Note these lines are merely reindented under a `with changed_dir(self.build_temp)` block.

**Comment by pitrou** [comment]
AppVeyor build at https://ci.appveyor.com/project/pitrou/arrow/build/1.0.10

**Comment by pitrou** [comment]
As usual, some `apt-get` failures on Travis-CI.

**Comment by wesm** [comment]
Restarted them. I also opened https://issues.apache.org/jira/browse/ARROW-2071 about reducing the runtime of builds

**Review by wesm — APPROVED** [review]
+1, thanks @pitrou!

**Pull request closed without merging**